source/src/ModuleApplication.cc

```diff
@@ -302,21 +349,47 @@ namespace dqm4hep {
       
       auto document = m_parser.document();
       const core::TiXmlHandle xmlHandle(document.RootElement());
-
       auto storageElement = xmlHandle.FirstChildElement("storage").Element();
+      auto settingsElement = xmlHandle.FirstChildElement("settings").Element();
+      auto moduleElement = xmlHandle.FirstChildElement("module").Element();
+      auto archiverElement = xmlHandle.FirstChildElement("archiver").Element();
+
       if(nullptr != storageElement) {
         THROW_RESULT_IF(core::STATUS_CODE_SUCCESS, !=, m_monitorElementManager->parseStorage<OnlineElement>(storageElement));
       }
+                  
+      if(nullptr == settingsElement) {
+        dqm_error("parseSteeringFile: Missing <settings> section !");
+        throw core::StatusCodeException(core::STATUS_CODE_NOT_FOUND);
+      }
       
-      auto moduleElement = xmlHandle.FirstChildElement("module").Element();
-      configureModule(moduleElement);
+      // determine running mode
+      static core::StringVector possibleModes = {"Online", "EventReader"}; 
```

Comment:
Maybe add a map with the mode enum to translate enum<->string to avoid having the `possibleModes` hardcoded here. Might be `problematic` if we decide to add a new mode later.